core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1856,9 +1867,22 @@ class SparkContext(config: SparkConf) extends Logging {
       throw new SparkException(s"Cannot stop SparkContext within listener bus thread.")
     }
     // Use the stopping variable to ensure no contention for the stop scenario.
-    // Still track the stopped variable for use elsewhere in the code.
+    if (!stopping.compareAndSet(false, true)) {
+      if (stopped.get()) {
+        logInfo("SparkContext already stopped.")
+      } else {
+        logInfo("SparkContext is being stopped.")
+      }
+      return
+    }
+    stopHooks.synchronized {
+      stopHooks.forEach {
+        hook => Try(Utils.logUncaughtExceptions(hook()))
+      }
+    }
     if (!stopped.compareAndSet(false, true)) {
-      logInfo("SparkContext already stopped.")
+      // this should not happen
+      logError("SparkContext already stopped.")
```

Comment:
Same message with different log level might confuse us. If we would like to leave error message because it should not reach here, explaining situation would not hurt since its level is ERROR.

Reply:
Changed to info so that messages are printed at the same log level from different places.